### PR TITLE
C-33941

### PR DESCRIPTION
**Pull request by Kulkarni-Kaustubh**
Signed-off-by: Kaustubh Kulkarni <kaustubh.j.kulkarni@seagate.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [x] Attach test execution logs
-  [x] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: tests/prov/test_multiple_config_deploy.py [hunk 1]
```diff
@@ -60,6 +61,7 @@ def setup_class(cls):
                 cls.master_node_list.append(node_obj)
             else:
                 cls.worker_node_list.append(node_obj)
+        cls.log_disk_size = os.getenv('log_disk_size')
```
Comment (pragamj) [inline tests/prov/test_multiple_config_deploy.py:64]: for this we need to make change in Jenkins job

**Comment by pragamj** [comment]
Please add collection.log, execution logs

**Comment by Kulkarni-Kaustubh** [comment]
[Test_45452.txt](https://github.com/Seagate/cortx-test/files/9409013/Test_45452.txt)
[Test_45453.txt](https://github.com/Seagate/cortx-test/files/9409039/Test_45453.txt)
[Test_31390.txt](https://github.com/Seagate/cortx-test/files/9412682/Test_31390.txt)
[Test-45450.txt](https://github.com/Seagate/cortx-test/files/9412684/Test-45450.txt)

**Comment by Kulkarni-Kaustubh** [comment]
Logs are attach

File: libs/prov/prov_k8s_cortx_deploy.py [hunk 2]
```diff
@@ -533,9 +533,9 @@ def update_sol_yaml(self, worker_obj: list, filepath: str, cortx_image: str,
                 assert False, "The requested data disk is more than" \
                               " the data disk available on the system"
             if log_disk_flag:
-                # the '1' is being added to accumulate the final data disk count
+                # the '2' is being added to accumulate the final data disk count
                 # after excluding the log disks
-                data_devices_f = device_list[cvg_count+1:]
+                data_devices_f = device_list[cvg_count+2:]
```
Comment (pragamj) [inline libs/prov/prov_k8s_cortx_deploy.py:538]: this will not work in case of data_disk_per_cvg =0

Comment (pragamj) [reply libs/prov/prov_k8s_cortx_deploy.py:538]: use data_devices_f = device_list[cvg_count*2:]

**Comment by Kulkarni-Kaustubh** [comment]
[Test_31390-0dc.txt](https://github.com/Seagate/cortx-test/files/9432765/Test_31390-0dc.txt)
[Test_45453-0ldc.txt](https://github.com/Seagate/cortx-test/files/9432766/Test_45453-0ldc.txt)

This log file are by keeping data_disk_per_cvg: 0

File: tests/prov/test_multiple_config_deploy.py (hunk 1 above)
Comment (Kulkarni-Kaustubh) [reply tests/prov/test_multiple_config_deploy.py]: Remove and keep the log disk size 4G.

File: libs/prov/prov_k8s_cortx_deploy.py [hunk 3]
```diff
@@ -533,9 +533,9 @@ def update_sol_yaml(self, worker_obj: list, filepath: str, cortx_image: str,
                 assert False, "The requested data disk is more than" \
                               " the data disk available on the system"
             if log_disk_flag:
-                # the '2' is being multiplied to accumulate the final data disk count
-                # after excluding the metadata,log disks
-                data_devices_f = device_list[cvg_count*2:]
+                # the '2' is being added to accumulate the final data disk count
+                # after excluding the log disks
+                data_devices_f = device_list[cvg_count+2:]
```
Comment (pragamj) [inline libs/prov/prov_k8s_cortx_deploy.py:538]: use the Multiplication instead of addition

File: libs/prov/prov_k8s_cortx_deploy.py [hunk 4]
```diff
@@ -581,7 +581,7 @@ def update_sol_yaml(self, worker_obj: list, filepath: str, cortx_image: str,
         if not resource_resp:
             assert False, "Failed to update the resources for thirdparty"
         # Update resources for cortx component
-        cortx_resource_resp = self.update_res_limit_cortx(filepath, cvg_count)
+        cortx_resource_resp = self.update_res_limit_cortx(filepath)
```
Comment (pragamj) [inline libs/prov/prov_k8s_cortx_deploy.py:584]: don't overwrite the method while pulling from main

File: tests/prov/test_multiple_config_deploy.py [hunk 5]
```diff
@@ -95,9 +95,8 @@ def multiple_node_deployment(self, node, config):
             dix_parity=config['dix_parity'], dix_spare=config['dix_spare'],
             cvg_count=config['cvg_per_node'], data_disk_per_cvg=config['data_disk_per_cvg'],
             master_node_list=self.master_node_list, worker_node_list=self.worker_node_list,
-            s3_instance=1)
+            s3_instance=2, log_disk_flag=log_device)
```
Comment (pragamj) [inline tests/prov/test_multiple_config_deploy.py:98]: keep s3 instances =1

File: tests/prov/test_multiple_config_deploy.py [hunk 6]
```diff
@@ -72,20 +72,20 @@ def teardown_method(self):
             support_bundle_utils.collect_support_bundle_k8s(local_dir_path=path,
                                                             scripts_path=
                                                             self.deploy_conf['k8s_dir'])
-        if self.destroy_flag:
```
Comment (pragamj) [inline tests/prov/test_multiple_config_deploy.py:75]: same here keep the changes from main

**Review by pragamj — APPROVED** [review]

File: config/prov/deploy_config.yaml [hunk 7]
```diff
@@ -63,6 +63,15 @@ nodes_3:
     dix_spare : 0
     cvg_per_node : 3
     data_disk_per_cvg : 1
+  config_6:
```
Comment (priyankaborawake) [inline config/prov/deploy_config.yaml:66]: We have a similar config defined in config_2. 
Can you please explain why this is needed. 
cc @pragamj

File: config/prov/deploy_config.yaml [hunk 8]
```diff
@@ -175,6 +184,33 @@ nodes_5:
...
+  config_8:
+    sns_data: 8
+    sns_parity: 2
+    sns_spare: 0
+    dix_data: 1
+    dix_parity: 2
+    dix_spare: 0
+    cvg_per_node: 2
+    data_disk_per_cvg: 1
+  config_9:
```
Comment (priyankaborawake) [inline config/prov/deploy_config.yaml:205]: Similar is defined in config_2

File: libs/prov/prov_k8s_cortx_deploy.py (hunk 2 above)
Comment (priyankaborawake) [inline libs/prov/prov_k8s_cortx_deploy.py:538]: @Kulkarni-Kaustubh As per pragam's comment, should multiplication be used here ?

File: tests/prov/test_multiple_config_deploy.py [hunk 9]
```diff
@@ -95,7 +98,11 @@ def multiple_node_deployment(self, node, config):
             dix_parity=config['dix_parity'], dix_spare=config['dix_spare'],
             cvg_count=config['cvg_per_node'], data_disk_per_cvg=config['data_disk_per_cvg'],
             master_node_list=self.master_node_list, worker_node_list=self.worker_node_list,
+<<<<<<< HEAD
```
Comment (priyankaborawake) [inline tests/prov/test_multiple_config_deploy.py:101]: @Kulkarni-Kaustubh The code is not properly merged. 
Please check.

File: libs/prov/prov_k8s_cortx_deploy.py [hunk 10]
```diff
@@ -533,8 +533,13 @@ def update_sol_yaml(self, worker_obj: list, filepath: str, cortx_image: str,
                 assert False, "The requested data disk is more than" \
                               " the data disk available on the system"
             if log_disk_flag:
+<<<<<<< HEAD
```
Comment (pragamj) [inline libs/prov/prov_k8s_cortx_deploy.py:536]: please remove these HEAD and other unwanted lines.

File: libs/prov/prov_k8s_cortx_deploy.py [hunk 11]
```diff
@@ -533,6 +533,8 @@ def update_sol_yaml(self, worker_obj: list, filepath: str, cortx_image: str,
                 assert False, "The requested data disk is more than" \
                               " the data disk available on the system"
             if log_disk_flag:
+                # the '2' is being added to accumulate the final data disk count
+                # after excluding the log disks
```
Comment (pragamj) [inline libs/prov/prov_k8s_cortx_deploy.py:537]: remove the comments as its redundant

File: tests/prov/test_multiple_config_deploy.py [hunk 12]
```diff
@@ -78,14 +79,15 @@ def teardown_method(self):
...
-    def multiple_node_deployment(self, node, config):
+    def multiple_node_deployment(self, node, config, **kwargs):
         """
         This Method is used for deployment of various node count
         and its multiple SNS,DIX configs
         :param: nodes: Its the count of worker nodes in K8S cluster.
         :param: config: Its the config for each node defined
                         in deploy_config.yaml file
         """
+        log_device = kwargs.get("log_device_flag", None)
```
Comment (pragamj) [inline tests/prov/test_multiple_config_deploy.py:90]: make this None to False

**Review by priyankaborawake — APPROVED** [review]

**Comment by priyankaborawake** [comment]
@Kulkarni-Kaustubh Please attach the latest collection logs. 
Approved PR.

**Review by ravindrachoudhari — APPROVED** [review]

**Review by pragamj — APPROVED** [review]

**Review by ketanarlulkar — APPROVED** [review]

**Pull request merged**